Assets/MixedRealityToolkit-SDK/Features/Boundary/System/MixedRealityBoundarySystem.cs

```diff
@@ -0,0 +1,491 @@
+        public override void Register(GameObject listener)
+        {
+            base.Register(listener);
+        }
+
+        /// <summary>
+        /// UnRegisters the <see cref="GameObject"/> to listen for boundary events.
+        /// /// </summary>
+        /// <param name="listener"></param>
+        public override void Unregister(GameObject listener)
+        {
+            base.Unregister(listener);
+        }
+
+        #endregion
+
+        #region IMixedRealityEventSource Implementation
+
+        /// <inheritdoc />
+        bool IEqualityComparer.Equals(object x, object y)
+        {
+            // There shouldn't be other Boundary Managers to compare to.
+            return false;
+        }
+
+        /// <inheritdoc />
+        public int GetHashCode(object obj)
+        {
+            return Mathf.Abs(SourceName.GetHashCode());
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; } = 0;
+
+        /// <inheritdoc />
+        public string SourceName { get; } = "Mixed Reality Boundary System";
+
+        #endregion IMixedRealityEventSource Implementation
+
+        #region IMixedRealityBoundarySystem Implementation
+
+        /// <inheritdoc/>
+        public ExperienceScale Scale { get; set; }
+
+        /// <inheritdoc/>
+        public float BoundaryHeight { get; set; } = 3f;
+
+        // todo: coming in Beta
+        //// Default to not showing the boundary walls.
+        // private bool showBoundaryWallss = false;
+        ///// <inheritdoc/>
+        //public bool ShowBoundaryWalls { get; set; } = false;
+
+        // todo: coming in Beta
+        //// Default to not showing the ceiling plane.
+        // private bool showCeiling = false;
+        ///// <inheritdoc/>
+        //public bool ShowCeiling { get; set; } = false;
+
+        // Default to showing the floor plane.
+        private bool showFloor = true;
+
+        /// <inheritdoc/>
+        public bool ShowFloor
+        {
+            get { return showFloor; }
+            set
+            {
+                if (showFloor != value)
+                {
+                    showFloor = value;
+
+                    if (value && (currentFloorPlane == null))
+                    {
+                        GetFloorVisualization();
+                    }
+
+                    currentFloorPlane?.SetActive(value);
+
+                    RaiseBoundaryVisualizationChanged();
+                }
+            }
+        }
+
+        private bool showPlayArea = true;
+
+        /// <inheritdoc/>
+        public bool ShowPlayArea
+        {
+            get { return showPlayArea; }
+            set
+            {
+                if (showPlayArea != value)
+                {
+                    showPlayArea = value;
+
+                    if (value && (currentPlayArea == null))
+                    {
+                        GetPlayAreaVisualization();
+                    }
+
+                    currentPlayArea?.SetActive(value);
```

Comment:
These operators do not use the custom equality operators declared on `UnityEngine.Object`, and so bypass a check to see if the underlying native Unity engine object has been destroyed. An explicit null comparison is preferred in order to clarify intent.